src/map/map.ts

```diff
@@ -347,92 +313,114 @@ export class PropertiesMap {
         this._settingsModal.remove();
     }
 
-    public get active(): string {
-      if (this._selected.size > 0) {
+    /**
+     * Change the environment indicated by the currently active marker to
+     * the one with the given `indexes`
+     */
+    public select(indexes: Indexes) {
         if (this._active === undefined) {
-            this.active = this._selected.keys().next().value;
-            return this._selected.keys().next().value;
-        } else {
-            return this._active; }
-      } else {
-            return ''; }
+            throw Error('tries to update selected environment, but there is no active marker');
+        }
+
+        const data = this._selected.get(this._active);
+        assert(data !== undefined);
+
+        // Plotly.restyle fires the plotly_click event, so ensure we only run
+        // the update once.
+        // https://github.com/plotly/plotly.js/issues/1025
+        if (data.current === 0 || indexes.environment !== data.current) {
+            data.current = indexes.environment;
+            // Sets the active marker on this map
+            this._updateSelectedMarker(data);
+        }
     }
+
     /**
-     * Function to set the active marker for communicating with the structure
-     * viewer
+     * Set the marker with given GUID as the active marker.
      *
-     * @param  activeGUID the GUID of the new active viewer
+     * @param guid the GUID of the new active viewer
      */
-    public set active(activeGUID: string) {
-        if (this._active === activeGUID) {
-            // guard against infinite recursion
-            // `Structure set active => StructureViewer.show => Map set active
-            // => PropertiesMap.select => Structure set active`
-            return;
+    public setActive(guid: GUID) {
```

Comment:
Why did you change from the `set active` style?

Reply:
For two reason: we sometimes want to use `this._active = '...'` and sometimes `this.active = '...'`. Having a setter instead of a function make the code confusing, since you have to remember the difference. It also allow to follow the same style as the rest of the actions on the map/structure grid: `map.setActive(...)`, `map.select(...)`, etc.